run-tester.sh

```diff
@@ -2,26 +2,83 @@
 
 set -euo pipefail
 
+# This URL is used for firefox and chrome, and is only a placeholder for safari.
 # When run in macOS, the test harness does not use WebDriver because
-# SafariDriver's "glass pane" feature interferes with testing. Provide a valid
-# URL simply as a placeholder.
-url_placeholder=http://example.com
+# SafariDriver's "glass pane" feature interferes with testing.
+webdriver_url=http://127.0.0.1:4444
 
-aria-at-automation-driver/package/bin/at-driver serve --port 3031 > at-driver.log 2>&1 &
-
-atdriver_pid=$!
+# Initialize so we can set up trap right away
+webdriver_pid=0
+atdriver_pid=0
 
 function clean_up {
-  kill -9 ${atdriver_pid} || true
+  if [[ ${webdriver_pid} -ne 0 ]]; then
+    kill -9 ${webdriver_pid} || true
+  fi
+  if [[ ${atdriver_pid} -ne 0 ]]; then
+    kill -9 ${atdriver_pid} || true
+  fi
 }
 trap clean_up EXIT
 
+aria-at-automation-driver/package/bin/at-driver serve --port 3031 > at-driver.log 2>&1 &
+atdriver_pid=$!
+
+poll_url() {
+  local url="$1"
+  local attempt=0
+  echo "Polling ${url}"
+
+  while [ ${attempt} -lt 30 ]; do
+    ((attempt++))
+
+    response=$(curl -s -o /dev/null -v -w "%{http_code}" -m 2 "$url" || true)
```

Comment:
This seems to be printing out a bunch of communications logging via some debug output somewhere: https://github.com/bocoup/aria-at-gh-actions-helper/actions/runs/10047859862/job/27770538071#step:22:14  - it might be nice to only show this extremely verbose (and expected to be failing/useless) output when the `$RUNNER_DEBUG` environment is non-zero (this is a checkbox you can check when manually running a job)